jscomp/others/js_json.ml

```diff
@@ -60,6 +61,35 @@ let test (type a) (x : 'a) (v : a kind) : bool =
   | Array -> Js_array.isArray x 
   | Object -> (Obj.magic x) != Js.null && Js.typeof x = "object" && not (Js_array.isArray x )
 
-
 external parse : string -> t = "JSON.parse" [@@bs.val]
 (* TODO: more docs when parse error happens or stringify non-stringfy value *)
+
+let null : t = (Obj.magic (Js.null: 'a Js.null) : t)
+
+external string : string -> t = "%identity"
+
+external number : float -> t = "%identity"
+
+external numberOfInt : int -> t = "%identity"
+
+external boolean : Js.boolean -> t = "%identity" 
+
+let boolAsBoolean b = (Js_boolean.to_js_boolean b) |> boolean 
+
+external boolAsInt : bool -> t = "%identity" 
+
+external object_ : t Js_dict.t -> t = "%identity"
+
+external array_ : t array -> t = "%identity"
+
+external stringArray : string array -> t = "%identity"
+
+external numberArray : float array -> t = "%identity"
+
+external intArray : int array -> t = "%identity"
```

Comment:
same. since in JS  `int array` can be safely cast to `float array`, but I think such safe cast may find places in other module